src/main/java/fi/aalto/cs/apluscourses/utils/CoursesClient.java

```diff
@@ -147,6 +154,50 @@ public static void fetchAndConsume(@NotNull URL url,
     consumeResponseBody(request, consumer);
   }
 
+  /**
+   * Sends a POST request to the given URL.
+   *
+   * @param url            A URL.
+   * @param authentication The method of authentication.
+   * @param data           Map of request data.  Values can be strings, numbers or files.
+   *
+   * @throws IOException In case of I/O related errors or non-successful response.
+   */
+  public static void post(@NotNull URL url,
```

Comment:
"I strongly disagree with the" 
@OlliKiljunen @nikke234 

so my understanding of this class is: responsible for all the communication outside the plugin (course config & A+), and yet, the bunch of methods inside is not tested... my bad, I've overlooked this, but atm, my proposal is: either all the stuff inside is tested OR we research some ready-made third-party solutions that can at least partially cover the functionality. but after all, this **must** be tested (good question is by whom and when :smile: )

Reply:
The ready-made third-party solution here is Apache's HTTP client :smile: I agree that tests would be nice, although I don't personally think it's crucial they come in this PR